### DISCUSSION TITLE
led_set_brightness() is not setting brightness after led_blink() for STM32U575

### DISCUSSION THREAD

**Discussion by bhavikbhansali** [zephyrproject-rtos/zephyr#53925] [Q&A]
**Description**
I am trying to blink an LED using PWM. I came across the 2 inbuild function mentioned and used in [led_pwm example](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/led_pwm/src/main.c). This 2 functions are led_blink() and led_set_brightness(). When I am reading more about this function I came across one details which were mentioned is that if we call led_set_brightness() after led_blink() it will not affect led blinking. I am testing that and checked that it is not true. So just want to know and understand below things.

1. Is it something where if we call led_blink() function first and call led_set_brightness() after to that it will blink the led with led blink timing where led brightness is as per led_set_brightness() ?

2. Is it something where it is saying that if we call led_blink() for pin 1 and led_set_brightness() for  pin 2 it will not affect each other working?

image of the function
![image](https://user-images.githubusercontent.com/11437390/213352119-0dd9cff9-1609-4337-8dcd-60ca9757792f.png)

**Kindly Note that I am trying to use the same pin and calling the function one after another.**

 - Target platform on which I am trying it Nucleo U575ZI Q
 - What have you tried to diagnose or workaround this issue? No
- I am using Normal LED to test this.
- Zephyr 3.2.0

**To Reproduce**
1. Take a  [led_pwm example](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/led_pwm/src/main.c) from zephyr 3.2.0
2. Make sure appropriate board overlay is available
3. compile a code
4. run it and see the output

**Expected behavior**
As per understanding if we call led_blink() followed by led_set_brightness() led need to blink as per defined time and during the one time LED brightness need to be same as we define in led_set_brightness() function.

**Impact**
It is blocking my implemetition.

**Code**
In main I am using below code instead of actual example code.
`led_blink(lew_pwm, 0, 200, 800);`
`led_blink(lew_pwm, 1, 200, 800);`
`k_sleep(K_MSEC(5000));`
`led_set_brightness(lew_pwm, 0, 75);`
`led_set_brightness(lew_pwm, 1, 25);`

**Comment by FRASTM** [comment]
Seems there is a conflict in the DTS on the same pin being declared   (example of a disco_l475i_iot1)for the PWM output 
```
	pwm2: pwm {
		status = "okay";
		pinctrl-0 = <&tim2_ch1_pa5>;
	};
```

and being declared in the DTS of the board 
```
		green_led_1: led_1 {
			gpios = <&gpioa 5 GPIO_ACTIVE_HIGH>;
			label = "User LD1";
		};
```
--> this one should be removed when the pin pa5 is for pwm output

**Reply by FRASTM** [reply]
Please have a look at PR https://github.com/zephyrproject-rtos/zephyr/pull/54019

**Reply by bhavikbhansali** [reply]
Hello @FRASTM I check that for nucleo_u575zi_q and in my project configuration we are using only PWM no LED or GPIO.
 But still it is not working. Any other suggestion or update?
